File: rust/client/src/model/Recipe.js
import update from '../utils/deep-update';

export default class Recipe {
  id = null;
  name = '';
  description = '';
  complexity = 1; // [1; 10]
  popularity = 1; // [1; 10]
  ingredients = [];
  steps = [];
  hasImage = false;

  imageSrc = null;
  imageFile = null;

  constructor(data) {
    update(this, data);
  }
}

File: rust+yew/server/src/models/recipe.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Recipe {
    // ignore id for new recipe
    #[serde(skip_deserializing)]
    pub id: i32,
    pub name: String,
    pub description: String,
    pub complexity: i16,
    pub popularity: i16,
    pub has_image: bool,
    pub ingredients: serde_json::Value,
    pub steps: serde_json::Value,
}
